From 2cef4fb1d12e537466ecd6eaf0187a5c71662d71 Mon Sep 17 00:00:00 2001
From: Shawn Guo <shawn.guo@freescale.com>
Date: Fri, 13 Jun 2014 16:02:58 +0800
Subject: [PATCH] ENGR00317981: drm: set drm_device pointer into drvdata

If drvdata is not used by platform driver, let's set drm_device
pointer into it.  We take this as the default usage of drvdata,
and platform driver is free to overwrite it later as needed.

Signed-off-by: Shawn Guo <shawn.guo@freescale.com>
---
 drivers/gpu/drm/drm_platform.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/drm_platform.c b/drivers/gpu/drm/drm_platform.c
index 21fc820..dd9751c 100644
--- a/drivers/gpu/drm/drm_platform.c
+++ b/drivers/gpu/drm/drm_platform.c
@@ -53,6 +53,14 @@ static int drm_get_platform_dev(struct platform_device *platdev,
 
 	dev->platformdev = platdev;
 
+	/*
+	 * If drvdata is not used by platform driver, let's set drm_device
+	 * pointer into it.  We take this as the default usage of drvdata,
+	 * and platform driver is free to overwrite it later as needed.
+	 */
+	if (platform_get_drvdata(platdev) == NULL)
+		platform_set_drvdata(platdev, dev);
+
 	ret = drm_dev_register(dev, 0);
 	if (ret)
 		goto err_free;
-- 
1.8.0

